mlflow/projects/__init__.py

```diff
@@ -189,6 +194,31 @@ def _is_local_uri(uri):
     return not _GIT_URI_REGEX.match(uri)
 
 
+def _is_valid_version(repo, version):
+    """Verify the commit version of branch name is valid, raise exception if otherwise."""
+    from git.exc import GitCommandError
+    try:
+        return repo.git.rev_parse("--verify", version)
+    except GitCommandError:
+        # Second attempt to see if this is a branch
+        try:
+            return repo.git.rev_parse("--verify", "refs/remotes/origin/%s" % version)
+        except GitCommandError as e:
+            raise ExecutionException("Got exception when validating git version '%s' "
+                                     "- please ensure that the version exists in the repo. "
+                                     "Error: %s" % (version, e))
+
+
+def _is_valid_branch_name(uri, work_dir, branch):
+    """Returns True if the branch exists in the Git project."""
```

Comment:
Could we also update this docstring to state our assumption that `work_dir` is within a git repo?

Reply:
Done.